backend/api/communication/src/main/java/co/airy/core/api/communication/ConversationsController.java

```diff
@@ -168,6 +160,83 @@
         return ResponseEntity.noContent().build();
     }
 
+    @PostMapping("/conversations.addNote")
+    ResponseEntity<?> conversationAddNote(@RequestBody @Valid ConversationAddNoteRequestPayload payload) {
+        final String conversationId = payload.getConversationId().toString();
+        final String text = payload.getText();
+        if (text != null && (text.length() == 0 || text.length() >50000)) {
```

Comment:
You can also use annotations on the request payload class instead of writing this code. `@NotNull` and `@Size` should do the trick.